purchases/index.js

```diff
@@ -520,6 +520,22 @@ function listPurchases(db, options, callback) {
     });
 }
 
+function parseLimit({limit}){
+    const limitAsInt = parseInt(limit);
+    if(limitAsInt > 0 || !isNaN(limitAsInt)) {
```

Comment:
Logic error: you want limit to be "greater than 0 **and** not NaN". You have "Limit is greater than 0 or is a number" - which means 0 will also be accepted!

Reply:
You fixed this later